app/scripts/directives/createSecret.js

```diff
@@ -26,6 +26,11 @@ angular.module("openshiftConsole")
       templateUrl: 'views/directives/create-secret.html',
       link: function($scope) {
         $scope.nameValidation = DNS1123_SUBDOMAIN_VALIDATION;
+        $scope.secretReferenceValidation = {
+          pattern: /^[a-zA-Z0-9\-_]+$/,
+          minLength: 8,
+          description: 'Secret reference key must consist of lower-case, upper-case letters, numbers, bash and underscore.'
```

Comment:
dash not bash, also we use oxford commas here

"dash, and underscore"

Reply:
always forgetting about the commas. We dont use int with 'and' ... sorry